pocs/tests/test_pocs.py

```diff
@@ -343,3 +343,80 @@ def start_pocs():
 
     pocs_process.join()
     assert pocs_process.is_alive() is False
+
+
+def test_pocs_park_to_ready_with_obs(pocs):
```

Comment:
I'm worried that some of these states are getting too complex. Let's have a further discussion about how to address that (e.g. multiple state machines, either nested or parallel).

Reply:
Yes, this made the `parked` state a bit obtuse with all the `if`-branching.